doc/guides/env_vars.rst

```diff
@@ -0,0 +1,169 @@
+
+   # Windows
+   set MY_VARIABLE=foo
+
+.. warning::
+
+  This is best for experimentation. If you close your terminal window, use
+  another terminal window or tab, restart your computer, etc., this setting
+  will be lost forever.
+
+  Using options 2 or 3 is recommended if you want to keep using the setting.
+
+Option 2: In all Terminals
+--------------------------
+
+**macOS and Linux**:
+
+Add the ``export MY_VARIABLE=foo`` line to your shell's startup script in your
+home directory. For Bash, this is usually :file:`~/.bashrc` on Linux or
+:file:`~/.bash_profile` on macOS.  Changes in these startup scripts don't
+affect shell instances already started; try opening a new terminal window to get
+the new settings.
+
+**Windows**:
+
+You can use the ``setx`` program in ``cmd.exe`` or the third-party
+RapidEE program.
+
+To use ``setx``, type this command, then close the terminal window. Any new
+``cmd.exe`` windows will have :envvar:`MY_VARIABLE` set to ``foo``.
+
+.. code-block:: console
+
+   setx MY_VARIABLE foo
+
+To install RapidEE, a freeware graphical environment variable
+editor, `using Chocolatey`_ in an Administrator command prompt:
+
+.. code-block:: console
+
+   choco install rapidee
+
+You can then run ``rapidee`` from your terminal to launch the program and set
+environment variables. Make sure to use the "User" environment variables area
+-- otherwise, you have to run RapidEE as administrator. Also make sure to save
+your changes by clicking the Save button at top left before exiting.Settings
+you make in RapidEE will be available whenever you open a new terminal window.
+
+.. _env_vars_zephyrrc:
+
+Option 3: Using ``zephyrrc`` files
+----------------------------------
+
+Choose this option if you don't want to make the variable's setting available
+to all of your terminals, but still want to save the value for loading into
+your environment when you are using Zephyr.
+
+**macOS and Linux**:
+
+Create a file named :file:`~/.zephyrrc` if it doesn't exist, then add this line
+to it:
+
+.. code-block:: console
+
+   export MY_VARIABLE=foo
+
+To get this value back into your current terminal environment, **you must run**
+``source zephyr-env.sh`` from the main ``zephyr`` repository. Among other
+things, this script sources :file:`~/.zephyrrc`.
+
+The value will be lost if you close the window, etc.; run ``source
+zephyr-env.sh`` again to get it back.
+
+**Windows**:
+
+Add the line ``set MY_VARIABLE=foo`` to the file
+:file:`%userprofile%\\zephyrrc.cmd` using a text editor such as Notepad to save
+the value.
+
+To get this value back into your current terminal environment, **you must run**
+``zephyr-env.cmd`` in a ``cmd.exe`` window after changing directory to the main
+``zephyr`` repository.  Among other things, this script runs
+:file:`%userprofile%\\zephyrrc.cmd`.
+
+The value will be lost if you close the window, etc.; run ``zephyr-env.cmd``
+again to get it back.
+
+Option 4: Using Zephyr Build Configuration CMake package
```

Comment:
I think we should add a 5th option.

> Option 5: CMake pre-load script using `cmake -C <pre-load>.cmake`
>
> Chose this option if you have different sets of option you want to easily be able to choose between.
> This allows you to easily switch between a pre-defined set of options shared among multiple applications, for example:
> `cmake -Czephyr-sdk_release.conf` ... or `cmake -Cgnuarmemb_debug.cmake` which could then contain:
> ```
>  # zephyr-sdk_release.cmake
>  set(CONF_FILES=prj_release.conf CACHE STRING "Release conf file")
>  set(ZEPHYR_TOOLCHAIN_VARIANT zephyr CACHE STRING "Zephyr SDK")
> ```
> or 
> ```
> # gnuarmemb_debug.cmake
> set(CONF_FILES=prj_debug.conf CACHE STRING "Debug conf file")
> set(ZEPHYR_TOOLCHAIN_VARIANT gnuarmemb CACHE STRING "GNU arm embedded")
> ```

Reply:
I just noticed, the title of this page specifically says:
https://github.com/zephyrproject-rtos/zephyr/blob/b10beefc7573b105f90a26dfa1c99633d3040f63/doc/guides/env_vars.rst#environment-variables
> Environment Variables

The 4th Option of using a `Zephyr` build configuration package is not setting any environment variables.

So the title in that case is misleading.

Also CMake variables can be set using `-D<variable>=<value>` which is not mentioned at all.
This could give an impression that the best way to set `CONF_FILES` would be through one of those ways.

Note, not all Zephyr CMake variables are being sourced from environment setting.
For example `ZEPHYR_MODULES` is not, but surprisingly someone added `ZEPHYR_EXTRA_MODULES` here: https://github.com/zephyrproject-rtos/zephyr/pull/19301.
Similar with `BOARD_ROOT` and `SOC_ROOT`.

Reply:
> The 4th Option of using a `Zephyr` build configuration package is not setting any environment variables.
> 
> So the title in that case is misleading.

Ugh, this is worse than I thought :slightly_frowning_face: 

If you read the PR description, you will see that this content was originally in a page that was all about environment variables. Then someone got confused and mixed up cmake and environment variables and put the environment variable page into the application development page near the cmake variable setting content.

Then apparently at some point after that, the two really got mixed up even more.

Reply:
> If you read the PR description, you will see that this content was originally in a page that was all about environment variables. Then someone got confused and mixed up cmake and environment variables and put the environment variable page into the application development page near the cmake variable setting content.

but maybe we should take a step back, and instead describe how to set Zephyr build variables as that is what is important, and not whether it's done as environment variables.

And those can be set with the principles you already had located together in the first version, including setting those as an environment variable.

Reply:
> how to set Zephyr build variables as that is what is important, and not whether it's done as environment variables.

I disagree; the point of the page was to explain how to set environment variables.

Reply:
I disagree there.

We should describe common ways to set Zephyr build variables together, that is:
- `-D<var>=value`
- Environment one time: `export <var>=<val>` / `set <var>=<val>`
- All terminals: `~/.bashrc` / `~/.bash_profile` / `rapidee`
- Zephyr rc: `~/.zephyrrc` / `zephyrrc.cmd`
- Zephyr build configuration package.
- CMake pre-load cache: `-C <preload>.cmake`

That way the user can decide which principle is best suited to their need.

Having just environment variables in a dedicated page but the Zephyr build configuration package as a note, and not mentioning the `-C` preload option in any way may lead the user to believe the environment settings are the best or recommended way to go.

I think we should describe the possibilities together and let the reader determine what is best option.
Not scattering the different possibilities in various pages, making it harder to get the full picture on this.